docs(): Add local development instructions in CONTRIBUTING.md (#13651)
medusajs/medusa: CONTRIBUTING.md

3. Make your changes.
4. [Open a pull request towards the develop branch in the Medusa repo](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork). Within a couple of days a Medusa team member will review, comment and eventually approve your PR.

## Local development

> Prerequisites:
> 1. [Forked Medusa repository cloned locally](https://github.com/medusajs/medusa).
> 2. [A local Medusa application for testing](https://docs.medusajs.com/learn/installation).

```
|
|__ medusa  // forked repository
|
|__ test-project // medusa application for testing
|
|__ test-project_storefront // (optional) storefront to interact with medusa application
```

> > **Note:** Assuming your forked Medusa project and the test project are sibling directories, and you optionally setup the starter storefront as part of the installation, your directory structure would look like the above.

Reviewer: Would move this note to be above the directory structure.

```suggestion
The code snippets in this section assume that your forked Medusa project and the test project are sibling directories, and you optionally setup the starter storefront as part of the installation. For example:
```

    "@medusajs/locking-redis": "file:../medusa/packages/modules/providers/locking-redis",
    "@medusajs/locking-postgres": "file:../medusa/packages/modules/providers/locking-postgres",
    "@medusajs/telemetry": "file:../medusa/packages/medusa-telemetry",
    "@medusajs/settings": "file:../medusa/packages/modules/settings",
    "@medusajs/draft-order": "file:../medusa/packages/plugins/draft-order",
    "@medusajs/deps": "file:../medusa/packages/deps",
    "@medusajs/caching-redis": "file:../medusa/packages/modules/providers/caching-redis",
    "@medusajs/caching": "file:../medusa/packages/modules/caching"
}
```

2. Every time you make a change in the forked Medusa repository, you need to build the packages where the modifications took place with `yarn build`. Some packages have a watch script, so you can execute `yarn watch` once and it will automatically build on changes:

```bash
yarn build # or yarn watch
```

3. After building changes in the forked medusa repository, run the following command in the test project to regenerate the `node_modules` directory with the newly built contents from the previous step:

> ```

Reviewer: Add extra spacing before the code block

~~~suggestion

```
~~~

3. Make your changes.
4. [Open a pull request towards the develop branch in the Medusa repo](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork). Within a couple of days a Medusa team member will review, comment and eventually approve your PR.

## Local development

> Prerequisites:
> 1. [Forked Medusa repository cloned locally](https://github.com/medusajs/medusa).
> 2. [A local Medusa application for testing](https://docs.medusajs.com/learn/installation).

```
|
|__ medusa  // forked repository
|
|__ test-project // medusa application for testing
|
|__ test-project_storefront // (optional) storefront to interact with medusa application
```

> The code snippets in this section assume that your forked Medusa project and the test project are sibling directories, and you optionally setup the starter storefront as part of the installation. For example:

Reviewer: ~~~suggestion

The code snippets in this section assume that your forked Medusa project and the test project are sibling directories, and you optionally setup the starter storefront as part of the installation. For example:

```
|
|__ medusa  // forked repository
|
|__ test-project // medusa application for testing
|
|__ test-project_storefront // (optional) storefront to interact with medusa application
```

~~~

3. Make your changes.
4. [Open a pull request towards the develop branch in the Medusa repo](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork). Within a couple of days a Medusa team member will review, comment and eventually approve your PR.

## Local development

1. Provided you followed the previous section steps, you already have the forked Medusa repository locally and your branch created. Start making the changes you desire.

> 2. If you don't have already, create a local Medusa test project `npx create-medusa-app`. This is the project you are going to use to validate your changes at runtime.

Reviewer: ```suggestion
> Prerequisites:
> 1. [Forked Medusa repository cloned locally](https://github.com/medusajs/medusa).
> 2. [A local Medusa application for testing](https://docs.medusajs.com/learn/installation).
```

> 3. Replace the @medusajs/* dependencies and devDependencies in you test project package.json, to point to the corresponding local packages in your forked Medusa repository. You will also need to add the medusa packages in the resolutions section of the package.json, so that every dependency is resolved locally. Assuming your forked Medusa project lives inside the same directory as your test project (otherwise modify the paths accordingly) you would have:

Reviewer: ```suggestion
1. Replace the @medusajs/* dependencies and devDependencies in you test project's `package.json` to point to the corresponding local packages in your forked Medusa repository. You will also need to add the medusa packages in the resolutions section of the `package.json`, so that every dependency is resolved locally. For example, assuming your forked Medusa project and the test projects are sibling directories:
```

Reviewer: Can we add before the steps a small "diagram" of how the directory structure can look like? Something like this:

```
|
|__ medusa  // forked repository
|
|__ test-project

Author: Added a note also below the diagram

    "@medusajs/core-flows": "file:../medusa/packages/core/core-flows",
    "@medusajs/orchestration": "file:../medusa/packages/core/orchestration",
    "@medusajs/cli": "file:../medusa/packages/cli/medusa-cli",
    "@medusajs/modules-sdk": "file:../medusa/packages/core/modules-sdk",
    "@medusajs/workflows-sdk": "file:../medusa/packages/core/workflows-sdk",
    "@medusajs/framework": "file:../medusa/packages/core/framework",
    "@medusajs/auth-emailpass": "file:../medusa/packages/modules/providers/auth-emailpass",
    "@medusajs/locking-redis": "file:../medusa/packages/modules/providers/locking-redis",
    "@medusajs/locking-postgres": "file:../medusa/packages/modules/providers/locking-postgres",
    "@medusajs/telemetry": "file:../medusa/packages/medusa-telemetry",
    "@medusajs/settings": "file:../medusa/packages/modules/settings",
    "@medusajs/draft-order": "file:../medusa/packages/plugins/draft-order",
    "@medusajs/deps": "file:../medusa/packages/deps",
    "@medusajs/caching-redis": "file:../medusa/packages/modules/providers/caching-redis",
    "@medusajs/caching": "file:../medusa/packages/modules/caching"
}
```

> 4. Each time you make a change in the forked Medusa repository, you need to build the packages where the modifications took place with `yarn build`. Some packages have a watch script, so you can execute `yarn watch` once and it will automatically build on changes. 

Reviewer: ~~~suggestion
2. Every time you make a change in the forked Medusa repository, you need to build the packages where the modifications took place with `yarn build`. Some packages have a watch script, so you can execute `yarn watch` once and it will automatically build on changes:

```bash
yarn build # or yarn watch
```
~~~

> 5. To have this changes in your test project, you need to navigate to it and run `rm -R node_modules && yarn && yarn dev` so that node_modules is regenerated with the newly built contents from the previous step. 

Reviewer: ~~~suggestion
3. After building changes in the forked medusa repository, run the following command in the test project to regenerate the `node_modules` directory with the newly built contents from the previous step:

```
rm -R node_modules && yarn && yarn dev
```
~~~
